src/gcs-resumable-upload.ts

```diff
@@ -887,30 +820,6 @@ export class Upload extends Pumpify {
       this.offset = 0;
     } catch (e) {
       const err = e as GaxiosError;
-      const resp = err.response;
-      // we don't return a 404 to the user if they provided the resumable
-      // URI. if we're just using the configstore file to tell us that this
-      // file exists, and it turns out that it doesn't (the 404), that's
-      // probably stale config data.
-      if (
-        resp &&
-        resp.status === NOT_FOUND_STATUS_CODE &&
-        !this.uriProvidedManually
-      ) {
-        this.restart();
-        return;
-      }
-
-      // this resumable upload is unrecoverable (bad data or service error).
-      //  -
-      //  https://github.com/googleapis/gcs-resumable-upload/issues/15
-      //  -
-      //  https://github.com/googleapis/gcs-resumable-upload/pull/16#discussion_r80363774
-      if (resp && resp.status === TERMINATED_UPLOAD_STATUS_CODE) {
-        this.restart();
-        return;
-      }
-
```

Comment:
Why is this block getting removed?

Reply:
We don't want to retry 404s. This was the case of a stale URI. `TERMINATED_UPLOAD_STATUS_CODE` is not a retryable error code.